Quran/QuranPageCollectionViewCell.swift

```diff
@@ -50,5 +52,31 @@ class QuranPageCollectionViewCell: UICollectionViewCell {
             // add fit height
             sizeConstraints.append(mainImageView.addHeightConstraint(bounds.height - imageHeightDiff))
         }
+        mainImageView.layoutIfNeeded()
+
+        if let imageSize = mainImageView.image?.size {
+            let scale: CGFloat
+            if imageSize.width / imageSize.height < mainImageView.frame.size.width / mainImageView.frame.size.height {
+                scale = mainImageView.frame.size.height / imageSize.height
+            } else {
+                scale = mainImageView.frame.size.width / imageSize.width
+            }
+            let deltaX = (mainImageView.frame.size.width - (scale * imageSize.width)) / 2
+            let deltaY = (mainImageView.frame.size.height - (scale * imageSize.height)) / 2
+            highlightingView.setScaleInfo(scale, xOffset: deltaX, yOffset: deltaY)
+        }
+    }
+
+    func setAyahInfo(ayahInfoData: [AyahNumber: [AyahInfo]]?) {
+        self.ayahInfoData = ayahInfoData
+        highlightingView.ayahInfoData = ayahInfoData
+    }
+
+    func highlightAyah(ayah: AyahNumber) {
```

Comment:
I think `highlightAyah` should be `highlightAyahs` that takes a `Set<AyahNumber>`. Then we don't need to have setNeedsDisplay. Because we will be setting the ayahs and the `HighlightingView` will be responsible for the best way to redraw for example if we are using core animation we would use another method.

So, I suppose at the end this method should be something like:
```
func highlightAyahs(ayahs: Set<AyahNumber>) {
    highlightingView.highlightedAyat = ayahs
}
```